### PR TITLE
[WordSeg] Train until loss stops improving.

### PR DESCRIPTION
**Pull request by texasmichelle**
Remove fixed number of training epochs. Instead, continue training until validation loss has not improved for 5 consecutive epochs.

### CODE REVIEW

File: Examples/WordSeg/main.swift [hunk 1]
```diff
@@ -93,12 +95,24 @@ for epoch in 1...maxEpochs {
...
+
     print(
       """
       [Epoch \(epoch)] \
-      Training loss: \(trainingLossSum / Float(trainingBatchCount))
+      Training loss: \(trainingLoss))
       """
     )
+
+    // Stop training when loss stops improving.
```
Comment (sgugger) [inline Examples/WordSeg/main.swift:107]: The training loss should not be considered for the criterion, only the validation loss (they also only use the validation loss in the PyTorch implementation).

File: Examples/WordSeg/main.swift [hunk 2]
```diff
@@ -93,12 +95,24 @@ for epoch in 1...maxEpochs {
...
+    lossHistory.append(trainingLoss)
```
Comment (sgugger) [inline Examples/WordSeg/main.swift:108]: There should be two different arrays for training and validation loss as you can't compare them directly.

Comment (texasmichelle) [reply Examples/WordSeg/main.swift:108]: This will only be executed if there is no validation dataset. I added this to protect from runaway training times.

Comment (sgugger) [reply Examples/WordSeg/main.swift:108]: Ah! Didn't catch the guard above in the diff. It works then!
